x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/StartsWithFunctionProcessor.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.eql.expression.function.scalar.string;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xpack.eql.EqlIllegalArgumentException;
+import org.elasticsearch.xpack.ql.expression.gen.processor.Processor;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Objects;
+
+public class StartsWithFunctionProcessor implements Processor {
+
+    public static final String NAME = "sstw";
+
+    private final Processor source;
+    private final Processor pattern;
+
+    public StartsWithFunctionProcessor(Processor source, Processor pattern) {
+        this.source = source;
+        this.pattern = pattern;
+    }
+
+    public StartsWithFunctionProcessor(StreamInput in) throws IOException {
+        source = in.readNamedWriteable(Processor.class);
+        pattern = in.readNamedWriteable(Processor.class);
+    }
+
+    @Override
+    public final void writeTo(StreamOutput out) throws IOException {
+        out.writeNamedWriteable(source);
+        out.writeNamedWriteable(pattern);
+    }
+
+    @Override
+    public Object process(Object input) {
+        return doProcess(source.process(input), pattern.process(input));
+    }
+
+    public static Object doProcess(Object source, Object pattern) {
+        if (source == null) {
+            return null;
+        }
+        if (source instanceof String == false && source instanceof Character == false) {
+            throw new EqlIllegalArgumentException("A string/char is required; received [{}]", source);
+        }
+        if (pattern == null) {
+            return null;
+        }
+        if (pattern instanceof String == false && pattern instanceof Character == false) {
+            throw new EqlIllegalArgumentException("A string/char is required; received [{}]", pattern);
```

Comment:
nit: just a suggestion maybe extract the function to avoid repetition, for example:
https://github.com/elastic/elasticsearch/pull/54380/files#diff-5ad1821091fb7e7473cbdfdded44fbcbR52

Reply:
I believe that, at one point after all these string functions are done, we need to refactor them and extract some common code in a base class, if possible of course.